### PR TITLE
All claims 31 3 mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds Private Medical Records Uploads to All Claims 526 form.
Note: does not add anything 4142-related, as that will be added later

## Testing done
- Tested locally
- Added unit tests

## Screenshots
Base screen:
screen shot 2018-09-27 at 2 29 36 pm

-----

If user needs help they can click on the additional info tip:
screen shot 2018-09-27 at 2 29 46 pm

-----

If user selects 'no', they can just continue (since 4142 integration is not yet done):
screen shot 2018-09-27 at 2 30 10 pm

-----

If user selects 'yes', they can upload one or more documents:
screen shot 2018-09-27 at 2 30 17 pm

-----

They can edit the document name and type for each:
screen shot 2018-09-27 at 2 35 17 pm

## Acceptance criteria
- [x] Veteran can upload private medical records
- [x] Veteran can upload multiple private medical records
- [x] Veteran can choose to not upload private medical records

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by peggygannon — CHANGES_REQUESTED** [review]
@markgreenburg Looks good.. just a few content tweaks from me: 

1. Please add a period after this sentence:
Large files can take longer to upload with a slow Internet connection.

2. Please update the following buttons: 
--Upload >> Upload Document
--Add another record >> Add Another Document (with initial caps)

File: src/applications/disability-benefits/all-claims/constants.js [hunk 1]
```diff
@@ -134,6 +134,13 @@ export const SERVICE_CONNECTION_TYPES = {
 
 export const DATA_PATHS = {
   hasVAEvidence:
-    'view:hasEvidenceFollowUp.view:selectableEvidenceTypes.view:hasVAMedicalRecords'
+    'view:hasEvidenceFollowUp.view:selectableEvidenceTypes.view:hasVAMedicalRecords',
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/constants.js]: Oh hey, I didn't notice this before. Clever!

File: src/applications/disability-benefits/all-claims/pages/privateMedicalRecords.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    fileUploadUI(
+      'Upload your private medical records',
+      {
+        fileTypes: ['pdf', 'jpg', 'jpeg', 'png', 'gif', 'bmp', 'tif', 'tiff', 'txt'],
+        expandUnder: 'view:uploadPrivateRecordsQualifier',
+        expandUnderCondition: (data) => _.get(
+          'view:hasPrivateRecordsToUpload', data, false
+        ),
+        // TODO: This is the URL for Increase, check that it’s correct
+        fileUploadUrl: `${environment.API_URL}/v0/upload_supporting_evidence`,
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/pages/privateMedicalRecords.js]: @kreek Is this the right endpoint for uploading private medical records for all claims?

**Review by cvalarida — APPROVED** [review]
Looks great!

**Comment by markgreenburg** [comment]
@peggygannon here's a new screenshot with those 2 changes. Please take a look when you get a chance. Thanks!

screen shot 2018-10-02 at 10 08 59 am

**Review by peggygannon — APPROVED** [review]

**Pull request merged**